x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/SqlDataTypeConverter.java

```diff
@@ -95,22 +97,26 @@ public static DataType commonType(DataType left, DataType right) {
                 return right;
             }
         }
-        if (left == DATETIME) {
+        if (isDateTime(left)) {
             if (right == DATE || right == TIME) {
-                return left;
+                return DATETIME;
             }
             if (isInterval(right)) {
-                return left;
+                return DATETIME;
             }
         }
-        if (right == DATETIME) {
+        if (isDateTime(right)) {
             if (left == DATE || left == TIME) {
-                return right;
+                return DATETIME;
             }
             if (isInterval(left)) {
-                return right;
+                return DATETIME;
             }
         }
+        if (isDateTime(left) && isDateTime(right)) {
```

Comment:
DATETIME_NANOS and DATETIME are the same for all functionality apart from HitExtraction, so in such cases where we have operators on a pair of datetimes, doesn't have any difference. Hopefully all this will be removed once we have unified in an upcoming PR how we extract `date` and `date_nanos` for plain queries and aggregations.